### PR TITLE
MAC-71 Use MI authentication for payments events API

### PR DESCRIPTION
**Pull request by ChrisJWoodcock**

### CODE REVIEW

File: src/SFA.DAS.Forecasting.Application/Infrastructure/Registries/PaymentsRegistry.cs [hunk 1]
```diff
@@ -1,50 +1,22 @@
...
+            For<PaymentsEventsApiClientConfiguration>().Use(c => c.GetInstance<ForecastingConfiguration>().PaymentsEventsApi).Singleton();
+            For<IPaymentsEventsApiClientConfiguration>().Use(c => c.GetInstance<PaymentsEventsApiClientConfiguration>());
 
             if (ConfigurationHelper.IsDevEnvironment)
             {
                 For<IPaymentsEventsApiClient>().Use<DevPaymentsEventsApiClient>();
             }
-            else
-            {
-                For<IPaymentsEventsApiClient>().Use<PaymentsEventsApiClient>().Ctor<HttpClient>("httpClient").Is(c => CreateClient(c));
```
Comment (cofaulco) [inline src/SFA.DAS.Forecasting.Application/Infrastructure/Registries/PaymentsRegistry.cs:28]: where are we registering the actual PaymentsEventsApiClient?

**Review by cofaulco — COMMENTED** [review]
1 question

File: src/SFA.DAS.Forecasting.Application/Infrastructure/Registries/PaymentsRegistry.cs (hunk 1 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.Forecasting.Application/Infrastructure/Registries/PaymentsRegistry.cs:28]: It's in the base class of the registry

Comment (cofaulco) [reply src/SFA.DAS.Forecasting.Application/Infrastructure/Registries/PaymentsRegistry.cs:28]: should the dev registration calls 
`For<IPaymentsEventsApiClient>().ClearAll().Use<PaymentsEventsApiClient>().Ctor<HttpClient>("httpClient").Is(c => CreateClient(c));`

Comment (ChrisJWoodcock) [reply src/SFA.DAS.Forecasting.Application/Infrastructure/Registries/PaymentsRegistry.cs:28]: Fixed with ClearAll

**Review by cofaulco — APPROVED** [review]

**Comment by cofaulco** [comment]
closing as the upgrade to use MI with Provider Events API was done as part of NET6 work

**Pull request closed without merging**